modules/syslogformat/syslog-format.c

```diff
@@ -720,6 +720,10 @@ log_msg_parse_sd(LogMessage *self, const guchar **data, gint *length, const MsgF
         }
       while (left && open_sd != 0);
     }
+  else
+    {
+      goto error;
+    }
```

Comment:
This will break the parsing of RFC 5424 messages that have no SDATA fields and forgot to mark it with a NILVALUE (`-`).

This would not be a valid RFC 5424, so I guess it is okay to fall back to legacy parsing mode.

Reply:
you are right, this might have been intentional. I removing this from this PR.

Reply:
removed the functional change.

Reply:
I have now readded the stricter check when being invoked from sdata-parser() but at the same time relaxed it when doing SDATA parsing from RFC5424. This will match historical behaviour, as well as be strict when invoking it from sdata-parser() itself.

Also it makes the unit tests pass again. Let me know what you think.

Reply:
It is perfect. Thank you.